services/wallet/api.go

```diff
@@ -589,7 +591,67 @@ func (api *API) AddressExists(ctx context.Context, address types.Address) (bool,
 	return api.s.accountsDB.AddressExists(address)
 }
 
-// Returns details for the passed address (response doesn't include derivation path)
+// AddressDetails returns details for passed params (passed address, chains to check, timeout for the call to complete)
+// if chainIDs is empty, it will use all active chains
+// if timeout is zero, it will wait until the call completes
+// response doesn't include derivation path
+func (api *API) AddressDetails(ctx context.Context, params *requests.AddressDetails) (*DerivedAddress, error) {
+	if err := params.Validate(); err != nil {
+		return nil, err
+	}
+
+	result := &DerivedAddress{
+		Address: common.HexToAddress(params.Address),
+	}
+	addressExists, err := api.s.accountsDB.AddressExists(types.Address(result.Address))
+	if err != nil {
+		return result, err
+	}
+
+	result.AlreadyCreated = addressExists
+
+	chainIDs := params.ChainIDs
+	if len(chainIDs) == 0 {
+		activeNetworks, err := api.s.rpcClient.NetworkManager.GetActiveNetworks()
+		if err != nil {
+			return nil, err
+		}
+
+		chainIDs = wcommon.NetworksToChainIDs(activeNetworks)
+	}
+
+	clients, err := api.s.rpcClient.EthClients(chainIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx1 := ctx
+	if params.TimeoutInMilliseconds > 0 {
+		var cancel context.CancelFunc
+		ctx1, cancel = context.WithTimeout(ctx, time.Duration(params.TimeoutInMilliseconds)*time.Millisecond)
```

Comment:
Can probably just override the original timeout
```suggestion
		ctx, cancel = context.WithTimeout(ctx, time.Duration(params.TimeoutInMilliseconds)*time.Millisecond)
```